src/components/engine-connector/engine-connector.js

```diff
@@ -230,6 +230,7 @@ class EngineConnector extends Component {
     eventManager.listen(player, player.Event.AD_LOADED, e => {
```

Comment:
Isn't the issue exactly the event listener?
The ad loaded event - we don't populate the player with ad data so getAd() will return null.

Reply:
That's right

Reply:
so why not sync it to ad started?

Reply:
This will not fix the issue, this is just ad data props. The ad itself already initialized on ad started.

Reply:
only if we had getPendingAd API :-)